google-api-go-generator/replacements.go

```diff
@@ -4,31 +4,121 @@
 
 package main
 
+// deprecatedPkgs is a map that contains packages that should be
+// deprecated in favor of another package.
+type deprecatedPkgs map[string]*apiReplacement
+
+func (d deprecatedPkgs) Get(name, version string) string {
+	ar := d[name]
+	if ar == nil {
+		return ""
+	}
+	for _, v := range ar.versions {
+		if v == "*" || v == version {
+			return ar.replacementPkg
+		}
+	}
+	return ""
+}
+
 // replacementPackage is a map from an API package name to the
 // import path of the package that replaces it. If an API appears
 // in this map, its package doc comment will note that it is deprecated
 // and point to the replacement.
 // TODO(jba): consider automating this by looking at the structure of the gocloud repo.
-var replacementPackage = map[string]string{
-	"bigquery":          "cloud.google.com/go/bigquery",
-	"cloudkms":          "cloud.google.com/go/kms/apiv1",
-	"cloudtasks":        "cloud.google.com/go/cloudtasks/apiv2beta2",
-	"dataproc":          "cloud.google.com/go/dataproc/apiv1",
-	"datastore":         "cloud.google.com/go/datastore",
-	"dialogflow":        "cloud.google.com/go/dialogflow/apiv2",
-	"dlp":               "cloud.google.com/go/dlp/apiv2",
-	"firestore":         "cloud.google.com/go/firestore",
-	"language":          "cloud.google.com/go/language/apiv1",
-	"logging":           "cloud.google.com/go/logging",
-	"monitoring":        "cloud.google.com/go/monitoring/apiv3",
-	"oslogin":           "cloud.google.com/go/oslogin/apiv1",
-	"pubsub":            "cloud.google.com/go/pubsub",
-	"redis":             "cloud.google.com/go/redis/apiv1",
-	"spanner":           "cloud.google.com/go/spanner",
-	"speech":            "cloud.google.com/go/speech/apiv1",
-	"texttospeech":      "cloud.google.com/go/texttospeech/apiv1",
-	"translate":         "cloud.google.com/go/translate",
-	"videointelligence": "cloud.google.com/go/videointelligence/apiv1",
-	"vision":            "cloud.google.com/go/vision/apiv1",
-	"storage":           "cloud.google.com/go/storage",
+var replacementPackage deprecatedPkgs = map[string]*apiReplacement{
+	"bigquery": {
+		replacementPkg: "cloud.google.com/go/bigquery",
+		versions:       []string{"*"},
```

Comment:
I wonder if the default (no versions listed) should mean all?

Reply:
I like that idea, will do.